airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/dest_state_lifecycle_manager/DestSingleStateLifecycleManager.java

```diff
@@ -58,6 +59,22 @@ public void markFlushedAsCommitted() {
     }
   }
 
+  @Override
+  public void markCommittedAsEmitted() {
+    if (lastCommittedState != null) {
```

Comment:
This null check seems redundant; if lastCommitted is null, then this is a noop anyway

Reply:
Yes, I agree, the main rationale behind keeping the `null` check was that it follows the same pattern as the rest of `DestSingleStateLifecycleManager`